src/wazuh_modules/wm_syscollector.c

```diff
@@ -151,6 +152,15 @@ void* wm_sys_main(wm_sys_t *sys) {
         syscollector_start_ptr = so_get_function_sym(syscollector_module, "syscollector_start");
         syscollector_stop_ptr = so_get_function_sym(syscollector_module, "syscollector_stop");
         syscollector_sync_message_ptr = so_get_function_sym(syscollector_module, "syscollector_sync_message");
+
+        void* rsync_module = NULL;
+        if(rsync_module = so_check_module_loaded("rsync"), rsync_module) {
+            void (*rsync_initialize_ptr)(full_log_fnc_t) = so_get_function_sym(rsync_module, "rsync_initialize_log_function");
+            rsync_initialize_ptr(mt_log_wrapper);
+            #ifndef WIN32
+            so_free_library(rsync_module);
```

Comment:
Why only for non-win32 systems?

Reply:
This is related to the previous `so_check_module_loaded()` call.

In Linux, the flag `RTLD_NOLOAD` for `dlopen()` increments the internal counter anyway. We need a matching `dlclose()`. See https://stackoverflow.com/questions/50570223/dlopenrtld-noload-still-returns-not-null-after-dlclose

The documentation doesn't say anything about it
https://linux.die.net/man/3/dlopen

But here, we have a mention about it
https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/dlopen.3.html

```
     RTLD_NOLOAD     The specified image is not loaded.  However, a valid
                     handle is returned if the image already exists in the
                     process. This provides a way to query if an image is
                     already loaded.  The handle returned is ref-counted, so
                     you eventually need a corresponding call to dlclose()
```